packages/explat-react-client/src/index.ts

```diff
@@ -1,2 +1,50 @@
-// Stub index
-export default null;
+import { useState, useEffect } from 'react'
+
+import type { ExPlatClient, ExperimentAssignment } from '@automattic/explat-client'
+
+interface ExPlatReactClient {
+    /**
+     * An ExPlat useExperiment hook.
+     *
+     * NOTE: Doesn't obey ExperimentAssignment TTL in order to keep stable UX.
+     *
+     * @returns [isExperimentAssignmentLoading, ExperimentAssignment | null]
+     */
+    useExperiment: (experimentName: string) => [boolean, ExperimentAssignment | null]
+}
+
+export default function createExPlatReactClient(ExPlatClient: ExPlatClient): ExPlatReactClient {
+    return {
+        useExperiment: (experimentName: string) => {
+            const [previousExperimentName] = useState(experimentName)
+            const [isLoading, setIsLoading] = useState<boolean>(true);
+            const [
+                experimentAssignment,
+                setExperimentAssignment,
+            ] = useState<ExperimentAssignment | null>(null);
+
+            useEffect(() => {
+                let isSubscribed = true;
+                ExPlatClient.loadExperimentAssignment(experimentName).then((experimentAssignment) => {
+                    if (isSubscribed) {
+                        setExperimentAssignment(experimentAssignment);
+                        setIsLoading(false);
+                    }
+                });
+                return () => {
+                    isSubscribed = false;
+                };
+            }, []);
+
+            if (experimentName !== previousExperimentName) {
+                const message = '[ExPlat] useExperiment: experimentName should never change between renders!'
+                if ( ExPlatClient.internalUseOnlyIsDevelopmentMode ) {
```

Comment:
With a config attribute and a better variable name, this would be a bit easier to follow: `exPlatClient.config.isDevelopmentMode` is more informative.